csrc/dynamic_transform.cpp

```diff
@@ -668,9 +672,10 @@ void DynamicTransformConcretizer::mutate(TensorView* tv) {
         registerConcretization(out_id, concretized_out_id);
       }
 
-      // expr must be mutated in order to set it as the definition for the
-      // concretized outputs.
-      OptOutMutator::mutate(expr);
+      // Set expr as the definition for concretized outputs
```

Comment:
Please leave some more detailed comments why this needs to be done in this order.